extensions/ql-vscode/src/vscode-tests/cli-integration/global.helper.ts

```diff
@@ -93,6 +93,22 @@ export default function(mocha: Mocha) {
       removeStorage?.();
     }
   );
+
+  // check that the codeql folder is found in the workspace
+  (mocha.options as any).globalSetup.push(
+    async () => {
+      const folders = workspace.workspaceFolders;
+      if (!folders) {
+        fail('\n\n\nNo workspace folders found.\nYou will need a local copy of the codeql repo.\nMake sure you specify the path to it in launch.json.\nIt should be something along the lines of "${workspaceRoot}/../codeql" depending on where you have your local copy of the codeql repo.\n\n\n');
+      } else {
+        // look for the codeql basename in the workspace folders
+        const codeqlFolder = folders.find(folder => folder.name === 'codeql');
```

Comment:
I was thinking that you do a quick check for a specific file inside of the repository. Something like this instead:

```suggestion
        const codeqlFolder = folders.find(f => fs.pathExistsSync(path.join(f.uri.fsPath, 'codeql-workspace.yml'));
```

I haven't tried this out myself, so the syntax may be off.

Also, usually, we don't want to run synchronous functions in `fs`, but since this is just test setup I think we're fine.

Reply:
I'm afraid this doesn't seem to be working ** old-man-screams-at-file-paths **

I also tried: 

```
const codeqlFolder = folders.find(folder => folder.name === 'codeql');
const codeqlWorkspaceFile = fs.readdirSync(codeqlFolder.uri.fsPath).find(file => file.match(/^codeql-workspace\.yaml$/));
```

but no joy.

Reply:
How bad would it be if we just checked for the folder name?

Reply:
That's fine. Better than what we had before.